open-legends.lua

```diff
@@ -1,82 +1,44 @@
 -- open legends screen when in fortress mode
 --@ module = true
---[====[
 
-open-legends
-============
-Open a legends screen when in fortress mode. Requires a world loaded in fortress
-or adventure mode. Compatible with `exportlegends`.
-
-Note that this script carries a significant risk of save corruption if the game
-is saved after exiting legends mode. To avoid this:
-
-1. Pause DF
-2. Run `quicksave` to save the game
-3. Run `open-legends` (this script) and browse legends mode as usual
-4. Immediately after exiting legends mode, run `die` to quit DF without saving
-   (saving at this point instead may corrupt your save)
-
-Note that it should be safe to run "open-legends" itself multiple times in the
-same DF session, as long as DF is killed immediately after the last run.
-Unpausing DF or running other commands risks accidentally autosaving the game,
-which can lead to save corruption.
-
-The optional ``force`` argument will bypass all safety checks, as well as the
-save corruption warning.
-
-]====]
-
-local dialogs = require 'gui.dialogs'
-local gui = require 'gui'
-local utils = require 'utils'
-
-Wrapper = defclass(Wrapper, gui.Screen)
-Wrapper.focus_path = 'legends'
-
-local region_details_backup = {} --as:df.world_region_details[]
-
-function Wrapper:onRender()
-    self._native.parent:render()
-end
-
-function Wrapper:onIdle()
-    self._native.parent:logic()
+local dialogs = require('gui.dialogs')
+local gui = require('gui')
+local utils = require('utils')
+
+Restorer = defclass(Restorer, gui.Screen)
+Restorer.ATTRS{
+    focus_path='open-legends'
+}
+
+function Restorer:init()
+    print('initializing restorer')
+    self.region_details_backup = {} --as:df.world_region_details[]
+    local v = df.global.world.world_data.region_details
+    while (#v > 0) do
+        table.insert(self.region_details_backup, 1, v[0])
+        v:erase(0)
+    end
 end
 
-function Wrapper:onHelp()
-    self._native.parent:help()
+function Restorer:onIdle()
+    self:dismiss()
```

Comment:
Why are we calling dismiss() in onIdle()?

Reply:
that's it exactly -- the Restorer is underneath the Legends screen waiting to be exposed, at which point it would run once and dismiss itself. What you said in the other comment about the legends input logic for dismiss makes this not such a great idea, though.